packages/shared/consoleWithStackDev.js

```diff
@@ -36,14 +37,20 @@ export function error(format, ...args) {
   }
 }
 
+// eslint-disable-next-line react-internal/no-production-logging
+const supportsCreateTask = __DEV__ && enableOwnerStacks && !!console.createTask;
+
 function printWarning(level, format, args) {
   // When changing this logic, you might want to also
   // update consoleWithStackDev.www.js as well.
   if (__DEV__) {
     const isErrorLogger =
       format === '%s\n\n%s\n' || format === '%o\n\n%s\n\n%s\n';
 
-    if (ReactSharedInternals.getCurrentStack) {
+    if (!supportsCreateTask && ReactSharedInternals.getCurrentStack) {
+      // We only add the current stack to the console when createTask is not supported.
+      // Since createTask requires DevTools to be open to work, this means that stacks
+      // can be lost while DevTools isn't open but we can't detect this.
```

Comment:
Is this by design? Because I couldn't find any mentions of this on web

Assuming that logs are buffered, we should expect that users will end up having different stack traces, depending on if browser DevTools are opened or not?

Reply:
If the DevTools are not open, then there will only be the stack for the current callsite and not the "async stack". If that bottoms out inside React then effectively you have no stack trace.

This is not really new to `console.createTask` per se but even the built-in async stack traces in Chrome DevTools are only active if they DevTools is open. That's because they come with a cost even when you don't have an error. So if they were always on you'd effectively just slow down every site for every Chrome user - even if they're not a developer. Even for developers it'd slow down every page and not just the ones you're working on.

It's unfortunate for React because we don't actually bother calling `console.createTask` in production at all. So if everyone followed our pattern, it wouldn't slow down any sites except the "development mode" ones and you wouldn't have DevTools open.

User space dialogs would still be able to show stacks since they get our `new Error().stack` version.